src/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs

```diff
@@ -380,6 +380,29 @@ public void Match_Timeout_Throws()
             }).Dispose();
         }
 
+        [Theory]
+        [InlineData(RegexOptions.Compiled)]
+        [InlineData(RegexOptions.None)]
+        public void Match_Timeout_Loop_Throws(RegexOptions options)
+        {
+            var regex = new Regex(@"a\s+", options, TimeSpan.FromSeconds(1));
+            string input = @"a" + new string(' ', 800_000_000) + @"b";
+
+            Assert.Throws<RegexMatchTimeoutException>(() => regex.Match(input));
```

Comment:
Should we validate that the exception is thrown within some window of time?

Reply:
It would have to have significant leeway (eg., within 2x or 3x) to account for vagaries of CI machines.

Reply:
Oh, I see it's a second. Maybe within 30 sec?

Reply:
Assuming 800M will take much longer.

Reply:
30 seconds, 1 minute, whatever we think is reasonable.  My goal would just be that the test doesn't pass after the regex runs for an hour and then upon completion sees there was a timeout requested and throws :)